internal/display/display.go

```diff
@@ -0,0 +1,49 @@
+// SPDX-License-Identifier: Apache-2.0
+
+package display
+
+import (
+	"errors"
+	"io"
+	"os"
+	"os/exec"
+)
+
+func Display(defaultOutput, errorOutput io.Writer, contents []byte, page bool) error {
```

Comment:
I'd recommend structuring this a little bit differently. Instead of trying to pass through everything here, I'd have this func setup everything you need for a writer and return `(io.WriteCloser, error)`

```go
NewDisplayWriter(default io.Writer, page bool) (io.WriteCloser, error) {

if page {
  for _, bin := range []string{os.GetEnv("PAGER"), "less", "more"} {
    if _, err := exec.LookPath(bin); err != nil {
      continue
    }
    return execWriter(bin)
  }
}

return default

}
```

then where this is used:

```go
out, err := NewDisplayWriter(os.Stdout, true)
if err != nil {
  ...
}
defer out.Close()

fmt.Fprintln(out, ...)
```

This way it simplifies some of the parameters (e.g. should errors go to errorOutput or error?), and makes it easier to pass the writer around if we don't want to write all of the contents at once.

Reply:
I have refactored Aditya's version of this to work both in test, and in the command line.